content/client-lib-development-guide/features.textile

```diff
@@ -270,7 +272,9 @@ h3(#realtimeclient). RealtimeClient
 * @(RTC6)@ @RealtimeClient#time@ function:
 ** @(RTC6a)@ Proxy to @RestClient#time@ presented with an async or threaded interface as appropriate
 * @(RTC7)@ The client library must use the configured timeouts specified in the @ClientOptions@, falling back to the "client library defaults":#defaults and defaults described in @ClientOptions@ below
-* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth.authorise@ is called with @AuthOptions#force@ set to @true@ and the state is @CONNECTED@, @CONNECTING@, or @DISCONNECTED@, the client will obtain a new token, disconnect the current transport and resume the connection (see "@RTN15c@":#RTN15c for details on resume). A test should exist that performs an upgrade of capabilities without any message loss during the upgrade process. A test should also exist for a token change that fails due to an incompatible token, which should result in the connection entering the @FAILED@ state
+* @(RTC8)@ Authentication token changes for the current connection are possible when the client is in the @CONNECTED@, @CONNECTING@ or @DISCONNECTED@ states. If @auth#authorise@ is called or Ably requests a re-authentication (see "RTN22":#RTN22) and the state is @CONNECTED@, @CONNECTING@, or @DISCONNECTED@, the client will obtain a new token and send an @AUTH@ @ProtocolMessage@ to Ably. The @data@ attribute of the @ProtocolMessage@ must contain either a token string or an encoded @TokenDetails@ object with the @encoding@ attribute set accordingly
+* @(RTC8a)@ If the authentication token change is successful, then Ably will send a new @CONNECTED@ @ProtocolMessage@. The @connectionDetails@ provided in the @CONNECTED@ @ProtocolMessage@ must override any existing defaults, see "RTN21":#RTN21. A test should exist that performs an upgrade of capabilities without any loss of continuity or connectivity during the upgrade process. Another test should exist where the capabilities are downgraded resulting in Ably explicitly detaching the channel. That test must assert that the channel becomes detached soon after the token update and the reason is included in the channel state change event
```

Comment:
> Do we want to expose to the user that this has happened?

Can you think of a reason why someone would need to handle this event?  Assuming they cared they'd provide an authCallback, and if not, why should they care?  Happy to include if we can think of a use case.

> (does this spec item mean we may in the future be planning to expose capabilities in the connected message, btw?)

No, and I don't think we should.  We have to expect capabilities for a client could change in future without the client being involved, so I don't see any real benefit to the client knowing what capabilities it has as it could be incorrect anyway.